src/plugins/intel_cpu/src/cpu_tensor.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <memory>
+#include <string>
+#include <map>
+
+#include "cpu_tensor.h"
+#include "ie_blob.h"
+#include "ie_ngraph_utils.hpp"
+#include "openvino/runtime/iremote_tensor.hpp"
+#include "openvino/runtime/properties.hpp"
+
+namespace ov {
+namespace intel_cpu {
+
+//For the beginning probably one contructor should be enough
+Tensor::Tensor(MemoryPtr memptr) : m_memptr{memptr}, m_strides{}, need_update_strides{true} {
+    OPENVINO_ASSERT(m_memptr != nullptr);
+
+    // only support plain data format ncsp.
+    auto blocked_desc = m_memptr->getDescWithType<BlockedMemoryDesc>();
+    OPENVINO_ASSERT(blocked_desc, "intel_cpu::Tensor only support blocked memory.");
+    OPENVINO_ASSERT(blocked_desc->hasLayoutType(LayoutType::ncsp), "intel_cpu::Tensor only support memory with ncsp layout.");
+
+    m_element_type = InferenceEngine::details::convertPrecision(blocked_desc->getPrecision());
+    m_shape = blocked_desc->getShape().toPartialShape().to_shape();
+
+    if (!shape_size(m_shape)) m_memptr->getMemoryMngr()->resize(0);
+}
+
+void Tensor::set_shape(ov::Shape new_shape) { // this is not as simple as it looks
+// if the ov::Shape is defined, then call cloneWithNewDims to the Memory object memory desc
+// if ov::Shape is undefined (dynamic), we probably have to reintialize the internal Memory object
+// using plain memory format
+    auto& desc = m_memptr->getDesc();
+    const auto newdesc = desc.cloneWithNewDims(new_shape, true);
+    m_memptr->redefineDesc(newdesc);
+
+    m_shape = std::move(new_shape);
+    need_update_strides = true;
+}
+
+const ov::element::Type& Tensor::get_element_type() const { // trivial -> simply convert the precision type
+    return m_element_type;
+}
+
+const ov::Shape& Tensor::get_shape() const { //trivial -> convert cpu Shape to ov::Shape
+    return m_shape;
+}
+
+size_t Tensor::get_size() const { // trivial -> call to getElementsCount to Shape
+    auto& desc = m_memptr->getDesc();
+    return desc.getShape().getElementsCount();
+}
+
+size_t Tensor::get_byte_size() const { // trivial-> call getCurrentMemSize to memory desc
+    auto& desc = m_memptr->getDesc();
+    return desc.getCurrentMemSize();
+}
+
+const ov::Strides& Tensor::get_strides() const { // here I would ask you to investiage what solution may be better
```

Comment:
This is undecisive either, as per the reply of Ilya to my email.

Reply:
Every const method must be thread safe as it is assumed that it doesn't change anything. If it does, then thread safety must be insured.

Reply:
this get_strides method would definitely change m_strides if we don't update it in set_shape and let get_strides do it, unless we don't have this m_strides member variable. 
and as Ilya's comments, it is user's responsibility to call set_shape and get_strides synchronously. that means ov::Tensor doesn't guarantee thread-safe.

Then what can we do to this ITensor implementation? 
We could add a mutex lock to set_shape and get_strides. But it would not be satisfying as we hope set_shape lighte-weighted as it might be invoked per iteration.

@maxnick What's your idea on the trade-offs?

Reply:
> and as Ilya's comments, it is user's responsibility to call set_shape and get_strides synchronously. that means ov::Tensor doesn't guarantee thread-safe.
Yes, indeed. But I'm talking about synchronization between two`get_strides` calls, since this method is const. At the moment it is not safe to call `get_strides` to the same object simultaneously from two different threads. However, the user assumes such behavior to be safe since he calls a const method.
Thus we have two solutions here:
1. Add synchronization to the `get_strides` call.
2. Reset `m_strides` in the constructor and every set_shape call.
Every solution has its own tradeoffs. As a first step it looks easier to update the strides in the corresponding modifying methods.

Reply:
done.